packages/core/saved-objects/core-saved-objects-migration-server-internal/src/actions/delete_by_query.test.ts

```diff
@@ -0,0 +1,158 @@
+import { catchRetryableEsClientErrors } from './catch_retryable_es_client_errors';
+import { errors as EsErrors } from '@elastic/elasticsearch';
+import { elasticsearchClientMock } from '@kbn/core-elasticsearch-client-server-mocks';
+import { deleteByQuery } from './delete_by_query';
+import { createDeleteByQueryResponse } from './delete_by_query.mocks';
+
+jest.mock('./catch_retryable_es_client_errors');
+
+describe('deleteByQuery', () => {
+  const deleteQuery = {
+    bool: {
+      should: ['server', 'deprecated'].map((type) => ({
+        term: {
+          type,
+        },
+      })),
+    },
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('calls catchRetryableEsClientErrors when the promise rejects', async () => {
+    // Create a mock client that rejects all methods with a 503 status code response.
+    const retryableError = new EsErrors.ResponseError(
+      elasticsearchClientMock.createApiResponse({
+        statusCode: 503,
+        body: { error: { type: 'es_type', reason: 'es_reason' } },
+      })
+    );
+    const client = elasticsearchClientMock.createInternalClient(
+      elasticsearchClientMock.createErrorTransportRequestPromise(retryableError)
+    );
+
+    const task = deleteByQuery({
+      client,
+      indexName: '.kibana_8.0.0',
+      query: deleteQuery,
+      conflicts: 'proceed',
+      refresh: true,
+    });
+    try {
+      await task();
+    } catch (e) {
+      /** ignore */
+    }
+    expect(catchRetryableEsClientErrors).toHaveBeenCalledWith(retryableError);
+  });
+
+  it('calls `client.deleteByQuery` with the correct parameters', async () => {
+    const client = elasticsearchClientMock.createInternalClient(
+      Promise.resolve({ hits: { hits: [] } })
+    );
+
+    const task = deleteByQuery({
+      client,
+      indexName: '.kibana_8.0.0',
+      query: deleteQuery,
+      conflicts: 'proceed',
+      refresh: true,
+    });
+
+    await task();
+
+    expect(client.deleteByQuery).toHaveBeenCalledTimes(1);
+    expect(client.deleteByQuery).toHaveBeenCalledWith({
+      index: '.kibana_8.0.0',
+      query: deleteQuery,
+      refresh: true,
+      wait_for_completion: false,
+      conflicts: 'proceed',
+    });
+  });
+
+  it('resolves with `Either.right` if the delete task is successfully created', async () => {
+    const client = elasticsearchClientMock.createInternalClient(
+      Promise.resolve({
+        took: 147,
+        timed_out: false,
+        task: 1234,
+      })
+    );
+
+    const task = deleteByQuery({
+      client,
+      indexName: '.kibana_8.0.0',
+      query: deleteQuery,
+      conflicts: 'proceed',
+      refresh: true,
+    });
+
+    const result = await task();
+
+    expect(Either.isRight(result)).toBe(true);
+    expect((result as Either.Right<any>).right).toEqual({ taskId: '1234' });
+  });
+
+  it('resolves with `Either.left`, if the delete query fails', async () => {
+    const versionConflicts = [
```

Comment:
should be moved to the wait for delete by query action test

Reply:
Yeah that one still needs some work / refactor.